rosbridge_library/src/rosbridge_library/internal/message_conversion.py

```diff
@@ -290,28 +326,37 @@ def _to_inst(msg, rostype, roottype, clock=None, inst=None, stack=None):
     # Check if it's uint8[], and if it's a string, try to b64decode
     for binary_type, expression in ros_binary_types_list_braces:
         if expression.sub(binary_type, rostype) in ros_binary_types:
+            assert isinstance(msg, Sequence)
             return _to_binary_inst(msg)
 
     # Check the type for time or rostime
     if rostype in ros_time_types:
+        assert isinstance(msg, dict | str)
+        assert inst is None or isinstance(inst, TimeMsg | DurationMsg), (
+            "Expected Time or Duration instance"
+        )
         return _to_time_inst(msg, rostype, clock, inst)
 
     # Check to see whether this is a primitive type
     if rostype in ros_primitive_types:
+        assert isinstance(msg, primitive_types)
         return _to_primitive_inst(msg, rostype, roottype, stack)
 
     # Check whether we're dealing with a list type
-    if inst is not None and type(inst) in list_types:
+    if inst is not None and isinstance(inst, list_types):
+        assert isinstance(msg, list_types)
         return _to_list_inst(msg, rostype, roottype, clock, inst, stack)
 
     # Otherwise, the type has to be a full ros msg type, so msg must be a dict
     if inst is None:
         inst = ros_loader.get_message_instance(rostype)
 
+    assert isinstance(msg, dict)
+    assert isinstance(inst, ROSMessage)
```

Comment:
I think asserts are not favorable, as they will just raise an `AssertionError` instead of manually raising a `TypeError`.